src/connection.h

```diff
@@ -216,26 +263,85 @@ static inline ssize_t connSyncReadLine(connection *conn, char *ptr, ssize_t size
 }
 
 /* Return CONN_TYPE_* for the specified connection */
-static inline int connGetType(connection *conn) {
+static inline const char *connGetType(connection *conn) {
     return conn->type->get_type(conn);
 }
 
 static inline int connLastErrorRetryable(connection *conn) {
     return conn->last_errno == EINTR;
 }
 
-connection *connCreateSocket();
-connection *connCreateAcceptedSocket(int fd);
+/* Get address information of a connection.
+ * @remote works as boolean type to get local/remote address */
+static inline int connAddr(connection *conn, char *ip, size_t ip_len, int *port, int remote) {
+    if (conn && conn->type->addr) {
+        return conn->type->addr(conn, ip, ip_len, port, remote);
+    }
 
-connection *connCreateTLS();
-connection *connCreateAcceptedTLS(int fd, int require_auth);
+    return -1;
+}
+
+/* Format an IP,port pair into something easy to parse. If IP is IPv6
+ * (matches for ":"), the ip is surrounded by []. IP and port are just
+ * separated by colons. This the standard to display addresses within Redis. */
+static  inline int formatAddr(char *buf, size_t buf_len, char *ip, int port) {
```

Comment:
why was this moved from anet? Seems like the right place, this doesn't seem strictly related to connections.

Reply:
> why was this moved from anet? Seems like the right place, this doesn't seem strictly related to connections.

Hi, 
Formating IP and port into a string, this is quite common. This can be also used by RDMA in the next step. RDMA supports RoCEv2, the address is similar to the IPv4 style.